### DISCUSSION TITLE
[FEATURE REQUEST] Deprecate the ``__utils__`` dunder.

### DISCUSSION THREAD

**Discussion by s0undt3ch** [saltstack/salt#67529] [Feature Requests]
**Is your feature request related to a problem? Please describe.**

The ``__utils__`` dunder(salt loader) tries to provide easy access to Salt's utility modules.

Issues with this approach:
* Utility functions should not be available in dunders, they are utility functions.
* All utilities are compacted into `<module>.<funcname>`, which does not recurse into submodules.
* This kind of usage, is prone to memory leaks(because utilities were not coded to work within a salt loader context). See https://github.com/saltstack/salt/pull/62007 and https://github.com/saltstack/salt/pull/62021

Potential problems if we remove ``__utils__``:
* Utility functions aren't available in jinja templating for salt states.
  * They shouldn't, jinja uses it's own filters and Salt provides additional filters

**Describe the solution you'd like**
Completely remove support for ``__utils__``

**Describe alternatives you've considered**
None.

**Comment by OrangeDog** [comment]
If it's being removed in 3006 then it needs to be deprecated in 3005.

**Comment by OrangeDog** [comment]
There's a SEP for this: https://github.com/saltstack/salt-enhancement-proposals/pull/66

**Comment by s0undt3ch** [comment]
> If it's being removed in 3006 then it needs to be deprecated in 3005.

The deprecation period, sadly, needs to be longer.
